File: 20251203_MarkdownOrderedListItemConverter.py
** start of main.py **

def convert_list_item(markdown):
    markdown = markdown.strip()

    # Check for number eligibility
    if not markdown[0].isdigit():
        print("Invalid format")
        return "Invalid format"
    elif not markdown[1] == ".":
        print("Invalid format")
        return "Invalid format"
    
    # Get the proper text
    markdown = f"<li>{markdown[2:].strip()}</li>"

    print(markdown)
    return markdown


convert_list_item("1. My item") 
convert_list_item(" 1.  Another item")
convert_list_item("1 . invalid item")
convert_list_item("2. list item text")
convert_list_item(". invalid again")
convert_list_item("A. last invalid")

** end of main.py **

File: 20251203_MarkdownOrderedListItemConverter.js
** start of script.js **

function convertListItem(markdown) {
  markdown = markdown.trim();

  // Create function to check for digit
  function isDigit(char) {
    return /\d/.test(char)
  };

  //Check for number eligibility
  if (!isDigit(markdown[0])) {
    console.log("Invalid format")
    return("Invalid format")
  } else if (markdown[1] !== ".") {
    console.log("Invalid format")
    return("Invalid format")
  };

  // Strip the first two then clean
  markdown = `<li>${markdown.slice(2).trim()}</li>`;

  console.log(markdown);
  return(markdown)
}

convertListItem("1. My item")
convertListItem(" 1.  Another item")
convertListItem("1 . invalid item")
convertListItem("2. list item text")
convertListItem(". invalid again")
convertListItem("A. last invalid")



** end of script.js **

